app/assets/javascripts/admin/addon/components/admin-report.gjs

```diff
@@ -0,0 +1,793 @@
+  }
+
+  @action
+  applyFilter(id, value) {
+    let customFilters = this.args.filters?.customFilters || {};
+
+    if (typeof value === "undefined") {
+      delete customFilters[id];
+    } else {
+      customFilters[id] = value;
+    }
+
+    this.refreshReport({ filters: customFilters });
+  }
+
+  @action
+  refreshReport(options = {}) {
+    if (!this.args.onRefresh) {
+      return;
+    }
+
+    this.args.onRefresh({
+      type: this.model.type,
+      mode: this.currentMode,
+      chartGrouping: options.chartGrouping,
+      startDate:
+        typeof options.startDate === "undefined"
+          ? this.startDate
+          : options.startDate,
+      endDate:
+        typeof options.endDate === "undefined" ? this.endDate : options.endDate,
+      filters:
+        typeof options.filters === "undefined"
+          ? this.args.filters?.customFilters
+          : options.filters,
+    });
+  }
+
+  @action
+  exportCsv() {
+    const args = {
+      name: this.model.type,
+      start_date: this.startDate.toISOString(true).split("T")[0],
+      end_date: this.endDate.toISOString(true).split("T")[0],
+    };
+
+    const customFilters = this.args.filters?.customFilters;
+    if (customFilters) {
+      Object.assign(args, customFilters);
+    }
+
+    exportEntity("report", args).then(outputExportResult);
+  }
+
+  @action
+  onChangeMode(mode) {
+    this.currentMode = mode;
+    this.refreshReport({ chartGrouping: null });
+  }
+
+  @bind
+  fetchOrRender() {
+    if (this.report) {
+      this._renderReport(this.report, this.forcedModes, this.currentMode);
+    } else if (this.args.dataSourceName) {
+      this._fetchReport();
+    }
+  }
+
+  @bind
+  _computeReport() {
+    if (this.isDestroying || this.isDestroyed) {
+      return;
+    }
+
+    if (!this._reports || !this._reports.length) {
+      return;
+    }
+
+    // on a slow network _fetchReport could be called multiple times between
+    // T and T+x, and all the ajax responses would occur after T+(x+y)
+    // to avoid any inconsistencies we filter by period and make sure
+    // the array contains only unique values
+    let filteredReports = this._reports.uniqBy("report_key");
+    let foundReport;
+
+    const sort = (report) => {
+      if (report.length > 1) {
+        return report.findBy("type", this.args.dataSourceName);
+      } else {
+        return report;
+      }
+    };
+
+    if (!this.startDate || !this.endDate) {
+      foundReport = sort(filteredReports)[0];
+    } else {
+      const reportKey = this.reportKey;
+      foundReport = sort(
+        filteredReports.filter((report) =>
+          report.report_key.includes(reportKey)
+        )
+      )[0];
+
+      if (!foundReport) {
+        return;
+      }
+    }
+
+    if (foundReport.error === "not_found") {
+      this.showFilteringUI = false;
+    }
+
+    this._renderReport(foundReport, this.args.forcedModes, this.currentMode);
+  }
+
+  @bind
+  _renderReport(report, forcedModes, currentMode) {
```

Comment:
this method should probably take only the `report` arg and use `this.args.forcedModes` and `this.currentMode` directly

also no need for `@bind` I think